decentraland/unity-explorer
# feat: local scene dev landscape

Issue: https://github.com/decentraland/unity-explorer/issues/1656

* Re-enables the landscape for Local Scene Development.
* Modify worlds dynamic terrain generation to also run during local scene development (by reading the new 'local scene parcels' collection introduced in the `/about` response from the sdk runtime for local scenes
* Updated `ServerAbout` + `ServerConfiguration` to be able to read the new collection field from the `/about` response.
* Updated `IRealmData` and `RealmData` to transport the new local scene parcels collection for its usage during dynamic terrain landscape generation
* Updated `FixedScenePointers` + `LoadFixedPointersSystem` to support "non URN fixed scenes", as the local scenes don't have a urn identifier, we just get their occupied parcels.
* Updated `IGlobalRealmController`, `RealmController`, `RealmNavigator` so that the fixed scenes async await returns a `SceneEntityDefinition` directly, instead of a promise. Since promises to get the `SceneEntityDefinition` for "urn scenes" and for "pointer scenes" are different and are handled internally.

https://github.com/user-attachments/assets/5e4c0b0f-43d9-495a-9299-87a33cc21869

-------------------------------------------------

Related PRs

* https://github.com/decentraland/js-sdk-toolchain/pull/1013

-------------------------------------------------

### TEST INSTRUCTIONS

1. Clone/Download any scene; For example the [sdk7 template scene](https://github.com/decentraland/sdk7-scene-template)
2. Install the SDK's `next` package running `npm i @dcl/sdk@next` in the root folder of the scene.
3. Run `npm run start -- --explorer-alpha` to start the scene
4. Kill the launcher/explorer that will auto-open, but leave the server running in the console
5. Download the build from this PR
6. Open the build from this PR connecting it to the locally running scene ([WIKI steps for that](https://github.com/decentraland/unity-explorer/wiki/How-To#connecting-a-custom-build-to-the-scene), remember to put the correct scene position in the console command)
7. **Finally: confirm that the landscape is enabled and it is limited around the scene (with a couple of parcels margin) as if it were a world**
